### PR TITLE
Drop @types/pino from dependencies

### LINKED ISSUE #1321: `@types/pino` should not be in `dependencies`
Before you submit an issue we recommend you drop into the [Gitter community](https://gitter.im/fastify) or [Fastify Help](https://github.com/fastify/help) and ask any questions you have or mention any problems you've had getting started with Fastify.

**Please read this entire template before posting any issue. If you ignore these instructions
and post an issue here that does not follow the instructions, your issue might be closed,
locked, and assigned the `missing discussion` label.**

## To Reproduce

Paste your code here:

```js

```

## Expected behavior

Paste the results here:

```js

```

## Your Environment

- *node version*: 6,8,10
- *fastify version*: >=1.0.0
- *os*: Mac, Windows, Linux
- *any other relevant information*

### PR DESCRIPTION
**Pull request by cemremengu**
Closes #1321.

I don't think we should backport this to 1.x

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: fastify.d.ts [hunk 1]
```diff
@@ -149,7 +147,7 @@ declare namespace fastify {
   interface ServerOptions {
     ignoreTrailingSlash?: boolean,
     bodyLimit?: number,
-    logger?: pino.LoggerOptions | boolean,
+    logger?: boolean,
```
Comment (delvedor) [inline fastify.d.ts]: I think this should be `any`. 
```suggestion
    logger?: any,
```

File: fastify.d.ts [hunk 2]
```diff
@@ -270,7 +268,7 @@ declare namespace fastify {
    */
   interface FastifyInstance<HttpServer = http.Server, HttpRequest = http.IncomingMessage, HttpResponse = http.ServerResponse> {
     server: HttpServer
-    log: pino.Logger
+    log: any
```
Comment (mcollina) [inline fastify.d.ts]: This should have an interface with `info`, ` debug`  log methods.

File: fastify.d.ts [hunk 3]
```diff
@@ -56,6 +54,24 @@ declare namespace fastify {
     validation?: Array<ValidationResult>;
   }
 
+  /**
+   * Pino logger
+   */
+  interface PinoLogger {
```
Comment (mcollina) [inline fastify.d.ts]: I'd just call it `Logger`.

**Comment by cemremengu** [comment]
@mcollina will look into it, still need to add tests

**Comment by mcollina** [comment]
I'm -1 to adding anything else. If they need anything, it can be cast.

**Pull request merged**